crates/uv-python/src/py_launcher.rs

```diff
@@ -1,93 +1,111 @@
-/// ```
-static PY_LIST_PATHS: LazyLock<Regex> = LazyLock::new(|| {
-    // Without the `R` flag, paths have trailing \r
-    Regex::new(r"(?mR)^ -(?:V:)?(\d).(\d+)-?(?:arm)?\d*\s*\*?\s*(.*)$").unwrap()
-});
-
-/// Use the `py` launcher to find installed Python versions.
-///
-/// Calls `py --list-paths`.
-pub(crate) fn py_list_paths() -> Result<Vec<PyListPath>, Error> {
-    // konstin: The command takes 8ms on my machine.
-    let output = info_span!("py_list_paths")
-        .in_scope(|| Command::new("py").arg("--list-paths").output())
-        .map_err(|err| {
-            if err.kind() == std::io::ErrorKind::NotFound {
-                Error::NotFound
-            } else {
-                Error::Io(err)
+/// Find all Pythons registered in the Windows registry following PEP 514.
+pub(crate) fn registry_pythons() -> Result<Vec<RegistryPython>, windows_result::Error> {
+    let mut registry_pythons = Vec::new();
+    for root_key in [CURRENT_USER, LOCAL_MACHINE] {
+        let Ok(key_python) = root_key.open(r"Software\Python") else {
+            continue;
+        };
+        for company in key_python.keys()? {
+            // Reserved name according to the PEP.
+            if company == "PyLauncher" {
+                continue;
             }
-        })?;
+            let Ok(company_key) = key_python.open(&company) else {
+                // Ignore invalid entries
+                continue;
+            };
+            for tag in company_key.keys()? {
+                let tag_key = company_key.open(&tag)?;
 
-    // `py` sometimes prints "Installed Pythons found by py Launcher for Windows" to stderr which we ignore.
-    if !output.status.success() {
-        return Err(Error::StatusCode {
-            message: format!(
-                "Running `py --list-paths` failed with status {}",
-                output.status
-            ),
-            exit_code: output.status,
-            stdout: String::from_utf8_lossy(&output.stdout).trim().to_string(),
-            stderr: String::from_utf8_lossy(&output.stderr).trim().to_string(),
-        });
+                if let Some(registry_python) = read_registry_entry(&company, &tag, &tag_key) {
+                    registry_pythons.push(registry_python);
+                }
+            }
+        }
     }
 
-    // Find the first python of the version we want in the list
-    let stdout = String::from_utf8(output.stdout).map_err(|err| Error::StatusCode {
-        message: format!("The stdout of `py --list-paths` isn't UTF-8 encoded: {err}"),
-        exit_code: output.status,
-        stdout: String::from_utf8_lossy(err.as_bytes()).trim().to_string(),
-        stderr: String::from_utf8_lossy(&output.stderr).trim().to_string(),
-    })?;
+    // The registry has no natural ordering, so we're processing the latest version first.
+    registry_pythons.sort_by(|a, b| {
+        // Highest version first (reverse), but missing versions at the bottom (regular order).
```

Comment:
What's a missing version? A registered Python without a version?

Reply:
A `None`

Reply:
Makes sense, can you make that clear in the comment though? I didn't learn what we did with missing versions until further down.